src/dht/findpeer.js

```diff
@@ -17,10 +19,58 @@ module.exports = (send) => {
       opts = {}
     }
 
-    send.andTransform({
+    const handleResult = (res, callback) => {
+      // Inconsistent return values in the browser
+      if (Array.isArray(res)) {
+        res = res[0]
+      }
+
+      // Type 2 keys (inconsistencies between go core and js core)
+      if (res.Type !== 2 && res.type !== 2) {
+        const errMsg = `key was not found (type 2)`
+
+        return callback(errcode(new Error(errMsg), 'ERR_KEY_TYPE_2_NOT_FOUND'))
+      }
+
+      // inconsistencies between go core and js core
+      let id
+      let addrs
+
+      if (res.Responses) {
+        id = res.Responses[0].ID
+        addrs = res.Responses[0].Addrs
+      } else {
+        id = res.responses[0].id
+        addrs = res.responses[0].addrs
+      }
+
+      // inconsistencies js / go - go does not add `/ipfs/{id}` to the address
```

Comment:
Do we _need_ to fix this here?

Reply:
So, `go-ipfs` response contains only address. As a consequence, it will fail the following test:

[ipfs/interface-ipfs-core/js/src/dht/findpeer.js#L48](https://github.com/ipfs/interface-ipfs-core/blob/master/js/src/dht/findpeer.js#L48)

I can change the test in `interface-ipfs-core` and remove this if you feel that it is less hacky.

Reply:
What I'm concerned about is that we're making assumptions about the format of the address. What does go-ipfs return when it's a circulit address? Would that cause a false positive?

Ideally both implementations should return the same format! Do you know the reason why go-ipfs returns the addr without the peer ID? Perhaps for a smaller payload? It would be pretty easy for js-ipfs to decapsulate the peer id from the addrs wouldn't it?